src/model.py

```diff
@@ -300,6 +280,28 @@ def get_sampling_params_dict(self, params_json):
 
         return params_dict
 
+    def response_loop(self):
+        while True:
+            item = self._response_queue.get()
+            # To signal shutdown a None item will be added to the queue.
+            if item is None:
+                break
+            response_sender, response, response_flag = item
+            del item
+            try:
+                response_sender.send(response, response_flag)
+            except Exception as e:
+                self.logger.log_error(
+                    f"An error occurred while sending a response: {e}"
+                )
+            finally:
+                if response_flag == pb_utils.TRITONSERVER_RESPONSE_COMPLETE_FINAL:
+                    self.ongoing_request_count -= 1
+                    del response_sender
+                    if self._response_queue.empty():
+                        gc.collect()
```

Comment:
Can we do garbage collection on `None` item only? I'm a bit concerned with calling garbage collection for every request completion.

Reply:
> Can we do garbage collection on `None` item only?

The `None` item is only placed into the response queue when the `finalize()` function is called, but the `finalize()` function is called only if all the response sender(s) are destructed. In other words, we will only see the `None` item after the `response_sender`(s) are garbage collected.

I think the best solution now is to garbage collect if the response queue is empty, which means garbage collect happens only if there is no new response to be sent at the moment a request is completed, so it should not create too much overhead ideally.

Here is the frequency and duration of garbage collects happened during benchmarking:
```
I0722 21:40:56.330660 2595 grpc_server.cc:2463] "Started GRPCInferenceService at 0.0.0.0:8001"
I0722 21:40:56.330826 2595 http_server.cc:4692] "Started HTTPService at 0.0.0.0:8000"
I0722 21:40:56.383418 2595 http_server.cc:362] "Started Metrics Service at 0.0.0.0:8002"
I0722 21:41:30.977492 2595 model.py:302] "----- Garabage collect start -----"
I0722 21:41:31.072659 2595 model.py:304] "----- Garabage collect end -----"
I0722 21:41:55.523516 2595 model.py:302] "----- Garabage collect start -----"
I0722 21:41:55.618196 2595 model.py:304] "----- Garabage collect end -----"
I0722 21:42:20.098470 2595 model.py:302] "----- Garabage collect start -----"
I0722 21:42:20.193094 2595 model.py:304] "----- Garabage collect end -----"
I0722 21:42:44.672155 2595 model.py:302] "----- Garabage collect start -----"
I0722 21:42:44.766406 2595 model.py:304] "----- Garabage collect end -----"
I0722 21:43:09.278387 2595 model.py:302] "----- Garabage collect start -----"
I0722 21:43:09.372832 2595 model.py:304] "----- Garabage collect end -----"
I0722 21:43:33.794605 2595 model.py:302] "----- Garabage collect start -----"
I0722 21:43:33.889078 2595 model.py:304] "----- Garabage collect end -----"
I0722 21:43:58.329119 2595 model.py:302] "----- Garabage collect start -----"
I0722 21:43:58.422977 2595 model.py:304] "----- Garabage collect end -----"
I0722 21:44:22.812579 2595 model.py:302] "----- Garabage collect start -----"
I0722 21:44:22.906684 2595 model.py:304] "----- Garabage collect end -----"
```